libvast/src/type.cpp

```diff
@@ -2049,23 +2062,62 @@ record_type::field_view record_type::field(const offset& index) const noexcept {
 }
 
 size_t record_type::flat_index(const offset& index) const noexcept {
-  VAST_ASSERT(!index.empty(), "offset must not be empty");
-  auto result = size_t{0};
-  const auto* record = table().type_as_record_type_v0();
-  for (size_t i = 0; i < index.size(); ++i) {
+  VAST_ASSERT(!index.empty(), "index must not be empty");
+  auto flat_index = size_t{0};
+  auto current_index = offset{0};
+  auto history = detail::stack_vector<const fbs::type::record_type::v0*, 64>{
```

Comment:
The 64 is deliberately chosen to avoid allocations for this and the offset at the same level, with `offset` being an alias for `stack_vector<size_t, 64>`, and `sizeof(size_t) == sizeof(void*)`.

I think there's a comment about this in some other function where we use this, which I can probably just copy over.

Reply:
Ah, I didn't realize that the number is measured in bytes, and that it's just a soft upper bound.